lib/consensus.js

```diff
@@ -39,6 +41,36 @@ function isValidBlockHeader(newHeader, previousHeaders, network = 'mainnet') {
     && hasGreaterThanMedianTimestamp(newHeader, previousHeaders);
 }
 
+/**
+ * validates an array of transaction objects, an array of tx hashes
+ * or a single transaction object or its hash
+ * against a merkleblock and the local header chain
+ * @param {Object[]|string[]|Object|string} transactions
+ * @param {Object} rawMerkleBlock - a base64 encoded merkleblock
+ * @param {Object} headerChain - an instance of an spvchain
+ * @return {boolean}
+ */
+async function isValidTransaction(transactions, rawMerkleBlock, headerChain) {
+  let tx = [];
+  if (!Array.isArray(transactions)) {
+    if ((typeof transactions === 'string') || (transactions instanceof Transaction)) {
+      tx.push(transactions);
+    } else {
+      throw new Error('Please check type of transactions parameter');
+    }
+  } else {
+    tx = transactions.slice();
+  }
+
+  const merkleBlock = new MerkleBlock(Buffer.from(rawMerkleBlock, 'base64'));
+  const localHeader = await headerChain.getHeader(merkleBlock.header.hash);
+  if (!localHeader) {
+    return false;
+  }
+  return merkleProofs.validateTxProofs(merkleBlock, tx);
+}
+
 module.exports = {
   isValidBlockHeader,
+  isValidTransaction,
```

Comment:
Maybe plural form?

Reply:
areValidTransactions?
areValidTransactionHashes?
isValidTransactionArray?
isValidTransactionHashesArray?
what do you think?

Reply:
The first one is good!

Reply:
done